app/core/store/modules/seasonalLeague.js

```diff
@@ -70,7 +74,11 @@ export default {
         splitRankings.push(...state.globalRankings.globalTop.slice(0, 10))
         splitRankings.push({ type: 'BLANK_ROW' })
         splitRankings.push(...state.globalRankings.playersAbove)
-        splitRankings.push(state.mySession)
+        // This hack is due to a race condition where the server returns the player
+        // in the 4 above or 4 below. Thus we prevent player seeing duplicate of their result.
```

Comment:
To expand on this, the server is constantly re-scoring players and we fetch your session and the 4 above/below separately. Thus there was a chance that your session appears duplicated.

Reply:
Maybe at some point we look into fetching 5 below/above and showing 4, so that even if we have to subtract one, our leaderboard still has an extra one to fill in for it? And/or seeing if Redis can query in a way to do this in 1-2 shots instead of 2-3 shots (since the 4-1-4 logic was written for querying from Mongo)?